### ISSUE TITLE
Cann't modify service provider's response when it's protected by Spring Cloud OAuth2. 

### ISSUE THREAD

**Issue by waitshang** [zalando/problem-spring-web#557]
Cann't modify service provider's response when it's protected by Spring Cloud OAuth2. 
When I request a service with wrong Bear Token, it returns 
```json
{
    "error": "invalid_token",
    "error_description": "2ba1194e-ab3d-464b-877e-c2979c6b3b84"
}
```

**Comment by whiskeysierra** [comment]
Just to verify, have you followed these instructions?
https://github.com/zalando/problem-spring-web/tree/main/problem-spring-web#security

**Comment by waitshang** [comment]
> Just to verify, have you followed these instructions?
> https://github.com/zalando/problem-spring-web/tree/main/problem-spring-web#security

I have used problem-spring-web-starter artifact. Is it necessary to follow those instructions?

**Comment by whiskeysierra** [comment]
No, I'd hope that's enough. Can you provide a failing test case as a pull request or a demo project with the minimal steps needed to reproduce the issue?

**Comment by waitshang** [comment]
> No, I'd hope that's enough. Can you provide a failing test case as a pull request or a demo project with the minimal steps needed to reproduce the issue?

It seems that the problem-spring-web project can't resolve `org.springframework.security.oauth2.common.exceptions.OAuth2Exception`. 

[problem-spring-web-demo.zip](https://github.com/zalando/problem-spring-web/files/5626748/problem-spring-web-demo.zip)
